tools/actions/lint-release-proposal-commit-list.mjs

```diff
@@ -18,15 +18,34 @@ import { createInterface } from 'node:readline';
 const stdinLineByLine = createInterface(process.stdin)[Symbol.asyncIterator]();
 
 const changelog = await readFile(CHANGELOG_PATH, 'utf-8');
-const commitListingStart = changelog.indexOf('\n### Commits\n');
-const commitListingEnd = changelog.indexOf('\n\n<a', commitListingStart);
-const commitList = changelog.slice(commitListingStart, commitListingEnd === -1 ? undefined : commitListingEnd + 1)
-  // Checking for semverness is too expansive, it is left as a exercice for human reviewers.
+const sectionTitles = [
+  '### Commits',
+  '### Semver-Major Commits',
+  '### Semver-Minor Commits',
+  '### Semver-Patch Commits'
+];
+
+let commitList = '';
+
+for (const title of sectionTitles) {
+  const start = changelog.indexOf(`\n${title}\n`);
```

Comment:
This would pull the commits from the `.0.0` release for all releaases, which won't work. We need to differentiate the releases – i.e. if `\n### Commits\n` can be found, it's semver-minor or semver-patch, otherwise it's semver-major